server/monitor/src/main/java/org/apache/accumulo/monitor/rest/master/MasterResource.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.monitor.rest.master;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.accumulo.core.master.thrift.DeadServer;
+import org.apache.accumulo.core.master.thrift.MasterMonitorInfo;
+import org.apache.accumulo.core.master.thrift.TabletServerStatus;
+import org.apache.accumulo.core.util.AddressUtil;
+import org.apache.accumulo.monitor.Monitor;
+import org.apache.accumulo.monitor.rest.logs.DeadLoggerInformation;
+import org.apache.accumulo.monitor.rest.logs.DeadLoggerList;
+import org.apache.accumulo.monitor.rest.tservers.BadTabletServerInformation;
+import org.apache.accumulo.monitor.rest.tservers.BadTabletServers;
+import org.apache.accumulo.monitor.rest.tservers.DeadServerInformation;
+import org.apache.accumulo.monitor.rest.tservers.DeadServerList;
+import org.apache.accumulo.monitor.rest.tservers.ServerShuttingDownInformation;
+import org.apache.accumulo.monitor.rest.tservers.ServersShuttingDown;
+import org.apache.accumulo.server.master.state.TabletServerState;
+
+/**
+ *
+ * Responsible for generating a new Master information JSON object
+ *
+ * @since 2.0.0
+ *
+ */
+@Path("/master")
+@Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+public class MasterResource {
+  public static final String NO_MASTERS = "No Masters running";
+
+  /**
+   * Gets the MasterMonitorInfo, allowing for mocking frameworks for testability
+   */
+  protected MasterMonitorInfo getMmi() {
+    return Monitor.getMmi();
+  }
+
+  /**
+   * Generates a master information JSON object
+   *
+   * @return master JSON object
+   */
+  @GET
+  public MasterInformation getTables() {
+
+    MasterInformation masterInformation;
+
+    if (Monitor.getMmi() != null) {
+      String gcStatus = "Waiting";
+      String label = "";
+      if (Monitor.getGcStatus() != null) {
```

Comment:
Get this once and set a local variable. It's redundant to keep invoking `Monitor.getGcStatus()` over and over.